### PR TITLE
Adds a utility function for copying contents between DBs

### PR DESCRIPTION
**Pull request by thorbjoernl**
## Change Summary

Adds a utility function for copying contents between dbs, and tests that this works as expected in all directions for supported db formats.

## Checklist

* [x] Start with a draft-PR
* [x] The PR title is a good summary of the changes
* [x] PR is set to AeroTools and a tentative milestone
* [x] Documentation reflects the changes where applicable
* [x] Tests for the changes exist where applicable
* [x] Tests pass locally
* [x] Tests pass on CI
* [x] At least 1 reviewer is selected
* [x] Make PR ready to review

### CODE REVIEW

File: src/aerovaldb/aerovaldb.py [hunk 1]
```diff
@@ -213,13 +213,14 @@ async def put_glob_stats(
...
+    @async_and_sync
+    async def list_glob_stats(
         self,
         project: str,
         experiment: str,
         /,
         access_type: str | AccessType = AccessType.URI,
-    ) -> Generator[str, None, None]:
+    ) -> list[str]:
         """Generator that lists the URI for each glob_stats object.
```
Comment (heikoklein) [inline src/aerovaldb/aerovaldb.py:224]: docs still say "Generator"

File: src/aerovaldb/utils/copy.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        logger.info(f"Processing item {i} of {len(await source.list_all())}")
+        data = await source.get_by_uri(uri, access_type=AccessType.JSON_STR)
+        await dest.put_by_uri(data, uri)
+
+    dst_len = len(await dest.list_all())
+    src_len = len(await source.list_all())
+    if dst_len != src_len:
+        raise IOError(
+            f"Unexpected number of items in destination after copy. Expected {src_len}, got {dst_len}"
+        )
```
Comment (heikoklein) [inline src/aerovaldb/utils/copy.py:43]: can you create a installed small command-line utility `aerovaldb_cp` from this function. Please use argparse or typer for arguments and help.

**Review by heikoklein — CHANGES_REQUESTED** [review]
Looks good, but I would like to have in addition a command-line utility for copy.

Why did you change from Generators to async in this PR? It feels somehow unrelated?

File: src/aerovaldb/utils/copy.py (hunk 2 above)
Comment (thorbjoernl) [reply src/aerovaldb/utils/copy.py:43]: Is this something you want me to do as part of this PR, or should I create an issue for it?

**Comment by thorbjoernl** [comment]
Re: generator->list

Asyncio makes it really difficult to wait for a coroutine to finish within a (synchronous) function, so it is easier if everything is a coroutine if asyncio is used. The reason for the change is that I am skeptical that async generators will work well with the async_and_sync decorator.

**Review by heikoklein — APPROVED** [review]
Thanks, looks good.

**Pull request merged**